### PR TITLE
Regulating terminal for svc

### PR DESCRIPTION
**Pull request by sebalaig**
Backport for Java PR #1072 https://github.com/powsybl/powsybl-core/pull/1072/
Add Regulating terminal for SVC. Update in XML serialization.

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: include/powsybl/iidm/StaticVarCompensator.hpp [hunk 1]
```diff
@@ -67,6 +78,8 @@ class StaticVarCompensator : public Injection {
 
     std::vector<double> m_reactivePowerSetpoint;
 
+    std::reference_wrapper<Terminal> m_regulatingTerminal;
```
Comment (mathbagu) [inline include/powsybl/iidm/StaticVarCompensator.hpp:81]: The Generator implementation is not compliant with this one, but this one seems better. Could you create another pull request to fix the generator implementation?

File: src/iidm/StaticVarCompensatorAdder.cpp [hunk 2]
```diff
@@ -24,12 +24,15 @@ StaticVarCompensatorAdder::StaticVarCompensatorAdder(VoltageLevel& voltageLevel)
 StaticVarCompensator& StaticVarCompensatorAdder::add() {
     checkBmin(*this, m_bMin);
     checkBmax(*this, m_bMax);
+    checkRegulatingTerminal(*this, m_regulatingTerminal, getNetwork());
     checkSvcRegulator(*this, m_voltageSetpoint, m_reactivePowerSetpoint, m_regulationMode);
 
-    std::unique_ptr<StaticVarCompensator> ptrSvc = stdcxx::make_unique<StaticVarCompensator>(getNetwork(), checkAndGetUniqueId(), getName(), m_bMin, m_bMax, m_voltageSetpoint, m_reactivePowerSetpoint, *m_regulationMode);
+    std::unique_ptr<Terminal> terminalAdded = checkAndGetTerminal();
+    std::unique_ptr<StaticVarCompensator> ptrSvc = stdcxx::make_unique<StaticVarCompensator>(getNetwork(), checkAndGetUniqueId(), getName(), m_bMin, m_bMax, m_voltageSetpoint,
+        m_reactivePowerSetpoint, *m_regulationMode, m_regulatingTerminal ? m_regulatingTerminal.get() : *terminalAdded);
```
Comment (mathbagu) [inline src/iidm/StaticVarCompensatorAdder.cpp:32]: Could you extract the code to initialize the regulatingTerminal  and store the result in a local variable:
```
    std::unique_ptr<Terminal> ptrTerminal = checkAndGetTerminal();
    Terminal& regulatingTerminal = m_regulatingTerminal ? m_regulatingTerminal.get() : *ptrTerminal;
    std::unique_ptr<StaticVarCompensator> ptrSvc = stdcxx::make_unique<StaticVarCompensator>(getNetwork(), checkAndGetUniqueId(), getName(), m_bMin, m_bMax, m_voltageSetpoint,
        m_reactivePowerSetpoint, *m_regulationMode, regulatingTerminal);
    ...
   Terminal& terminal = svc.addTerminal(std::move(ptrTerminal));
```

File: src/iidm/converter/xml/NetworkXml.cpp [hunk 3]
```diff
@@ -201,7 +201,7 @@ Network NetworkXml::read(std::istream& is, const ImportOptions& options, const A
         } else if (context.getReader().getLocalName() == HVDC_LINE) {
             HvdcLineXml::getInstance().read(network, context);
         } else if (context.getReader().getLocalName() == EXTENSION) {
-            std::string id2 = context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue("id"));
+            std::string id2 = context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue(ID));
```
Comment (mathbagu) [inline src/iidm/converter/xml/NetworkXml.cpp:204]: Should be done in a separate commit.

File: src/iidm/converter/xml/StaticVarCompensatorXml.cpp [hunk 4]
```diff
@@ -49,7 +53,16 @@ StaticVarCompensator& StaticVarCompensatorXml::readRootElementAttributes(StaticV
...
 void StaticVarCompensatorXml::readSubElements(StaticVarCompensator& svc, NetworkXmlReaderContext& context) const {
     context.getReader().readUntilEndElement(STATIC_VAR_COMPENSATOR, [this, &svc, &context]() {
-        AbstractConnectableXml::readSubElements(svc, context);
+        if (context.getReader().getLocalName() == REGULATING_TERMINAL) {
+            IidmXmlUtil::assertMinimumVersion(STATIC_VAR_COMPENSATOR, REGULATING_TERMINAL, xml::ErrorMessage::NOT_SUPPORTED, IidmXmlVersion::V1_1(), context.getVersion());
+            std::shared_ptr<std::string> id = std::make_shared<std::string>(context.getAnonymizer().deanonymizeString(context.getReader().getAttributeValue(ID)));
+            std::shared_ptr<std::string> side = std::make_shared<std::string>(context.getReader().getOptionalAttributeValue(SIDE, ""));
+            context.addEndTask([id, side, &svc]() {
+                svc.setRegulatingTerminal(stdcxx::ref<Terminal>(TerminalRefXml::readTerminalRef(svc.getTerminal().getVoltageLevel().getSubstation().getNetwork(), *id, *side)));
+            });
```
Comment (mathbagu) [inline src/iidm/converter/xml/StaticVarCompensatorXml.cpp:62]: I prefer to avoid shared_ptr: copying the string should be better.

File: src/iidm/converter/xml/StaticVarCompensatorXml.cpp [hunk 5]
```diff
@@ -63,6 +76,13 @@ void StaticVarCompensatorXml::writeRootElementAttributes(const StaticVarCompensa
     writePQ(svc.getTerminal(), context.getWriter());
 }
 
+void StaticVarCompensatorXml::writeSubElements(const StaticVarCompensator& svc, const VoltageLevel& /*voltageLevel*/, NetworkXmlWriterContext& context) const {
+    IidmXmlUtil::assertMinimumVersionAndRunIfNotDefault(!stdcxx::areSame(svc, svc.getRegulatingTerminal().getConnectable().get()),
+        STATIC_VAR_COMPENSATOR, REGULATING_TERMINAL, xml::ErrorMessage::NOT_DEFAULT_NOT_SUPPORTED, IidmXmlVersion::V1_1(), context.getVersion(), [&svc, &context]() {
```
Comment (mathbagu) [inline src/iidm/converter/xml/StaticVarCompensatorXml.cpp:81]: Is it really necessary to prefix ErrorMessage using xml?

Comment (sebalaig) [reply src/iidm/converter/xml/StaticVarCompensatorXml.cpp:81]: No it's not
